crates/holochain_p2p/src/spawn/actor.rs

```diff
@@ -941,13 +945,24 @@ impl HolochainP2pActor {
         let (s, r) = tokio::sync::oneshot::channel();
         self.pending.lock().unwrap().register(msg_id, s);
 
+        let start = std::time::Instant::now();
         space.send_notify(to_url, req).await?;
 
         match r.await {
             Err(_) => Err(HolochainP2pError::other(format!(
                 "{tag} response channel dropped: likely response timeout"
             ))),
-            Ok(resp) => cb(resp),
+            Ok(resp) => {
+                self.request_duration_metric.record(
+                    start.elapsed().as_secs_f64(),
+                    &[opentelemetry_api::KeyValue::new(
+                        "dna_hash",
+                        format!("{:?}", dna_hash),
+                    )],
+                );
+
+                cb(resp)
+            }
```

Comment:
Suggest adding `tag` and `url` as a distinguishers since the individual request types could vary wildly in how long they commonly take to process by the remote, and individual remotes could be very slow or fast.

- Do we also want to peek into the resp to see if it is an error type and record that?
- Do we also want to record a metric in the `Err` arm for timeouts?

Reply:
Good points, this can be much better -> https://github.com/holochain/holochain/pull/4835/commits/bb71241b93ef2999dadc74c7e8413ef8b23a7545

Reply:
woot :+1: